Update readme files with the necessary information to run the project (#12)
bleu-llc/balancer-pool-metadata: packages/contracts/README.md

# <h1 align="center"> Balancer Pool Metadata Contract</h1>

> This package contains the source code for the Balancer Pool Metadata smart-contract, the [`PoolMetadataRegistry`](./src/PoolMetadataRegistry.sol).

Reviewer: ```suggestion
This package contains the source code for the Balancer Pool Metadata Smart Contract, under `src/PoolMetadataRegistry.sol:PoolMetadataRegistry`.
```

# Getting started

> ## Filling environment variables

Reviewer: ```suggestion
## Environment variables
```

> To be able to test and deploy the smart contract, you need to create a `.env` file in the `contracts` directory. Check the [`.env.example`](./.env.template) for information.

Reviewer: ```suggestion
Make sure to create your own `.env` (`cp .env.template .env`) and add your variables before testing and trying to deploy the contract.
```

## Install dependencies

If you did not clone the repository using the `--recursive-submodules` option, make sure to install the submodules with

```bash

> $ git submodule update --init --recursive   # in the balancer-pool-metadata directory

Reviewer: `#question` does it make a difference if you run this outside of those directories?

```suggestion
$ git submodule update --init --recursive
```

Author: I tried to run it in the contract directory and it did not work

```

```bash
$ pnpm install
```

## Running tests

```bash
$ pnpm test
```

> ## Deploy to local

Reviewer 2: ```suggestion
## Deploy to local network
```

> We are using an anvil to create a local testnet node for deploying and testing smart contracts. Run the anvil server, get a generated private key in the terminal, and change your `PRIVATE_KEY` in the `.env`. Then deploy with the `deploy:local` command.

Reviewer: ```suggestion
We are using [anvil](https://book.getfoundry.sh/anvil/) to create a local testnet node for deploying and testing contracts. Run the anvil server, get a generated private key in the terminal, and change your `PRIVATE_KEY` in the `.env` file. Then deploy with the `deploy:local` command.
```

```bash
$ pnpm dev:anvil
$ pnpm deploy:local
```

> ## Deploy to goerli testnet

Reviewer 2: ```suggestion
## Deploy to Goerli testnet
```
